source/particle/property/elastic_stress.cc

```diff
@@ -0,0 +1,183 @@
+
+        data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_yy")));
+
+        if (dim == 2)
+          {
+            data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_xy")));
+          }
+        else if (dim == 3)
+          {
+            data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_zz")));
+
+            data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_xy")));
+
+            data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_xz")));
+
+            data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_yz")));
+          }
+
+      }
+
+      template <int dim>
+      void
+      ElasticStress<dim>::update_particle_property(const unsigned int data_position,
+                                                   const Vector<double> &solution,
+                                                   const std::vector<Tensor<1,dim> > &gradients,
+                                                   typename ParticleHandler<dim>::particle_iterator &particle) const
+      {
+        MaterialModel::MaterialModelInputs<dim> in(1, this->n_compositional_fields());
+
+        in.position[0] = particle->get_location();
+
+        in.current_cell = typename DoFHandler<dim>::active_cell_iterator(*particle->get_surrounding_cell(this->get_triangulation()),
+                                                                         &(this->get_dof_handler()));
+
+        in.temperature[0] = solution[this->introspection().component_indices.temperature];
+
+        in.pressure[0] = solution[this->introspection().component_indices.pressure];
+
+        for (unsigned int d = 0; d < dim; ++d)
+          in.velocity[0][d] = solution[this->introspection().component_indices.velocities[d]];
+
+        for (unsigned int n = 0; n < this->n_compositional_fields(); ++n)
+          in.composition[0][n] = solution[this->introspection().component_indices.compositional_fields[n]];
+
+        Tensor<2,dim> grad_u;
+        for (unsigned int d=0; d<dim; ++d)
+          grad_u[d] = gradients[d];
+        in.strain_rate[0] = symmetrize (grad_u);
+
+        MaterialModel::MaterialModelOutputs<dim> out(1, this->n_compositional_fields());
+        this->get_material_model().evaluate (in,out);
+
+        for (unsigned int i = 0; i < SymmetricTensor<2,dim>::n_independent_components ; ++i)
+          particle->get_properties()[data_position + i] += out.reaction_terms[0][i];
```

Comment:
Is it guaranteed that the stress terms are the first terms in these material models?

Reply:
Yes, that is enforced in the elastic rheology module.